distribution/lib/Standard/Base/0.0.0-dev/src/Data/Range.enso

```diff
@@ -1,11 +1,25 @@
-from Standard.Base import all
+import project.Data.Any.Any
+import project.Data.Filter_Condition.Filter_Condition
+import project.Data.Numbers.Integer
+import project.Data.Numbers.Number
+import project.Data.Vector.Vector
+import project.Function.Function
+import project.Nothing.Nothing
 
-# TODO Dubious constructor export
-from project.Data.Range.Range import all
-from project.Data.Range.Range export all
+from project.Data.Boolean import Boolean, True, False
+from project.Error.Common import Error, Illegal_Argument_Error_Data, Illegal_State_Error_Data
 
 ## Represents a right-exclusive range of integer values.
 type Range
+    ## Create a representation of a right-exclusive range of integer values.
+
+       Arguments:
+       - start: The left boundary of the range. Its value is included.
+       - end: The right boundary of the range. Its value is excluded.
+       - step: The step between values. Must be a positive value.
+    new : Integer -> Integer -> Integer -> Range
+    new start=0 end=100 step=1 =
+        Range.Value start end (if start > end then -1 else 1) . with_step step
```

Comment:
Currently `Range.new 1 10 2 . to_text` will yield `(Range.Value 1 10 2)`.
This is ok, but it is exposing the `.Value` constructor which I assume is meant to be internal now.
So I'm wondering if making it `Range.new 1 10 2` will not make it better.

Or maybe even making it `1.up_to 10 . with_step 2`. Seems like it would be much easier to read for the user what range is being represented. (We would need then to detect downward ranges and to `10.down_to 0 . with_step 4` and I guess it would be neat to have a special case for `step==1` where it is skipped so it is just `0.up_to 4`)

IMO this last approach would make for neatly looking representations, which would be valuable until we get some nice custom visualization for ranges (which I don't expect soon).